### ISSUE TITLE
browserdec.py - Microsoft Online Account

### ISSUE THREAD

**Issue by Banaanhangwagen** [tijldeneut/dpapilab-ng#1]
Placeholder to make `browserdec.py` compatible with a Microsoft Online Account.

Following this blogpost https://blog.nirsoft.net/2022/01/03/recover-dpapi-encrypted-passwords-created-under-microsoft-account-from-external-drive/
- not the user-password, but a random 44-characters password is used in the DPAPI-process
- this random pwd is cached and can be found in an encrypted form in `C:\Windows\System32\config\systemprofile\AppData\Local\Microsoft\Windows\CloudAPCache\MicrosoftAccount\[Account ID]\Cache\CacheData`

We just need to find out how to extract this.

**Comment by tijldeneut** [comment]
Hi, thank you for bringing it to my attention.
It's now on my to do list, I already figured out it only works with a Microsoft Online account that has no MFA configured (and also not with AzureAD, although there is a different folder for that).
Also the tool seems to not rely on third party DLL imports, which means any crypto is done in the EXE itself. I did figure out that the decrypted CacheData houses more than just the 32-byte random password. There is also an XML containing the BinaryDAToken probably for connecting to the .NET services.

Disassembling and debugging the EXE seems feasible enough, but finding the time is harder at the moment ;-)
